### PR TITLE
Improve stdout redirection in client process

### PR DESCRIPTION
**Pull request by Ashymad**
While researching https://github.com/JuliaEditorSupport/LanguageServer.jl/issues/330 I've found that the server waits for response from the client process indefinitely. This is because if there is an error while loading package in the :load_package case, the stdout is still redirected while the catch block executes, which causes the client process to print nothing and, in turn, for the  master to wait for response indefinitely. This PR fixes this and makes the code little bit nicer (IMO).

### CODE REVIEW

File: src/clientprocess/clientprocess_main.jl [hunk 1]
```diff
@@ -34,15 +34,12 @@ while true
             end
             serialize(stdout, (:success, pkgs))
         elseif message == :load_package
-            ostdout = stdout
-            (outRead, outWrite) = redirect_stdout() # seems necessary incase packages print on startup
-            SymbolServer.import_package(payload, depot)
+            redirect_stdout(open("/dev/null", "w")) do # seems necessary incase packages print on startup
```
Comment (davidanthoff) [inline src/clientprocess/clientprocess_main.jl]: `/dev/null` doesn't work on Windows...

Maybe just sticking with the old code but adding some `try finally` would also do the trick?

Comment (Ashymad) [reply src/clientprocess/clientprocess_main.jl]: Oh yeah, forgot about that... I see a few options:
 1. We could check the platform and use `"nul"` special file on Windows
 2. We could add inner try/catch/finally block in the `:load_package` case, rethrow the error in `catch` and redirect `stdout` in `finally`
 3. We could simply redirect `stdout` to `ostdout` in the catch block regardless if it was redirected earlier or not (don't know if not closing the pipe ends would be a problem)

Comment (davidanthoff) [reply src/clientprocess/clientprocess_main.jl]: I like the `nul` option, that seems cleanest!

**Comment by Ashymad** [comment]
Finally I've found some time to work on this. Now it should work on Windows as well

**Pull request merged**